File: Algorithms/Sorting/Selection_Sort/SelectionSort.java
// Java program for implementation of Selection Sort 

class SelectionSort 
{ 
	void sort(int arr[]) 
	{ 
		int n = arr.length; 

		for (int i = 0; i < n-1; i++) 
		{ 
			// Find the minimum element in unsorted array 
			int min_idx = i; 
			for (int j = i+1; j < n; j++) 
				if (arr[j] < arr[min_idx]) 
					min_idx = j; 

			// Swap the found minimum element with the first 
			// element 
			int temp = arr[min_idx]; 
			arr[min_idx] = arr[i]; 
			arr[i] = temp; 
		} 
	} 

	// Prints the array 
	void printArray(int arr[]) 
	{ 
		int n = arr.length; 
		for (int i=0; i<n; ++i) 
			System.out.print(arr[i]+" "); 
		System.out.println(); 
	} 

	// Driver code 
	public static void main(String args[]) 
	{ 
		SelectionSort obj = new SelectionSort(); 
		int arr[] = {50, 40, 30, 20, 10}; // Default Array
		obj.sort(arr);  // Calling sorting funtion
		obj.printArray(arr); //Printing Array
	} 
} 

File: Algorithms/Sorting/Selection_Sort/Selection_Sort.cpp
//Selection sort, sort the array by swapping the curr element with the smallest element from indexes curr+1 to len-1
#include<iostream>
using namespace std;
void selection_sort(int a[],int n){
    for(int i=0;i<n-1;i++){
        int min_index=i;
        for(int j=i+1;j<=n-1;j++){
            if(a[j]<a[min_index]){
                min_index=j;
            }
        }
      swap(a[i],a[min_index]);
    }
}
int main(){
    int n;
    cin>>n;
    int a[1000];
    for(int i=0;i<n;i++){
        cin>>a[i];
    }
    selection_sort(a,n);
    for(int i=0;i<n;i++){
        cout<<a[i]<<",";
    }
    return 0;
}
